### PR TITLE
chore: move metadata to pyproject.toml

### PR DESCRIPTION
**Pull request by SauravMaheshkar**
`pyproject.toml` is the new standard for packaging python packages (first introduced in [PEP 518](https://peps.python.org/pep-0518/) and later expanded in [PEP 517](https://peps.python.org/pep-0517/), [PEP 621](https://peps.python.org/pep-0621/) and [PEP 660](https://peps.python.org/pep-0660/)). This PR proposes to switch the metadata declaration from the legacy `setup.cfg` in favour of `pyproject.toml`. This would also allow for a better minimal project structure.

### CODE REVIEW

**Comment by SauravMaheshkar** [comment]
@nkoep Request for Review

File: pyproject.toml [hunk 1]
```diff
@@ -2,9 +2,97 @@
 requires = ["pip==21.3.1", "setuptools==59.6.0", "setuptools-scm[toml]>=6.2"]
 build-backend = "setuptools.build_meta"
 
+[project]
+name = "pymanopt"
+description = "Toolbox for optimization on Riemannian manifolds with support for automatic differentiation"
```
Comment (nkoep) [inline pyproject.toml:7]:
```suggestion
description = """\
  Toolbox for optimization on Riemannian manifolds with support for automatic \
  differentiation\
"""
```

File: pyproject.toml [hunk 2]
```diff
@@ -2,9 +2,97 @@
...
+authors = [
+  {name = "Jamie Townsend"},
+  {name = "Niklas Koep"},
+  {name = "Sebastian Weichwald"},
+]
+readme="README.md"
```
Comment (nkoep) [inline pyproject.toml:13]:
```suggestion
readme = "README.md"
```

File: pyproject.toml [hunk 3]
```diff
@@ -2,9 +2,97 @@
...
+license = { text = "BSD 3-Clause License"}
+requires-python=">=3.7"
+keywords=[
```
Comment (nkoep) [inline pyproject.toml:16]:
```suggestion
requires-python = ">=3.7"
keywords = [
```

**Review by nkoep — CHANGES_REQUESTED** [review]
Hey, thanks a lot for the PR! I've been meaning to get rid of the `setup.cfg` file as well. I just added a bunch of pedantic style comments. If you don't mind, could you also change all indents to 2 spaces for consistency? :bow:

File: pyproject.toml [hunk 4]
```diff
@@ -2,9 +2,97 @@
...
+[project]
+name = "pymanopt"
+description = "Toolbox for optimization on Riemannian manifolds with support for automatic differentiation"
+authors = [
+  {name = "Jamie Townsend"},
+  {name = "Niklas Koep"},
+  {name = "Sebastian Weichwald"},
+]
+readme="README.md"
+license = { text = "BSD 3-Clause License"}
```
[the remark this answers is not in the document]
Comment (SauravMaheshkar) [reply pyproject.toml]: Addressed in edda1b20cecb5a0b2ee767471ba284f301713545

File: pyproject.toml [hunk 5]
```diff
@@ -2,9 +2,97 @@
...
+    "License :: OSI Approved :: BSD License",
+    "Intended Audience :: Science/Research",
+    "Development Status :: 5 - Production/Stable",
+]
+dependencies = [
+    "numpy>=1.16.0",
+    "scipy>=1.0,<1.10",
+]
+[project.urls]
+homepage="https://pymanopt.org"
```
[the remark this answers is not in the document]
Comment (SauravMaheshkar) [reply pyproject.toml]: Addressed in edda1b20cecb5a0b2ee767471ba284f301713545

File: pyproject.toml (hunk 1 above)
Comment (SauravMaheshkar) [reply pyproject.toml]: Addressed in edda1b20cecb5a0b2ee767471ba284f301713545

File: pyproject.toml (hunk 2 above)
Comment (SauravMaheshkar) [reply pyproject.toml]: Addressed in edda1b20cecb5a0b2ee767471ba284f301713545

File: pyproject.toml (hunk 3 above)
Comment (SauravMaheshkar) [reply pyproject.toml]: Addressed in edda1b20cecb5a0b2ee767471ba284f301713545

**Comment by SauravMaheshkar** [comment]
> Hey, thanks a lot for the PR! I've been meaning to get rid of the `setup.cfg` file as well. I just added a bunch of pedantic style comments. If you don't mind, could you also change all indents to 2 spaces for consistency? 🙇

Hello @nkoep thanks a lot for the feedback. I addressed the requested changes in edda1b20cecb5a0b2ee767471ba284f301713545. Looking forward to contributing more 😄

**Review by nkoep — APPROVED** [review]
Thanks for the swift updates! LGTM!

**Comment by SauravMaheshkar** [comment]
@nkoep Do you think we should try and add a `pip install -U pip setuptools wheel` in the CI ? This repository does use the older version of `setup-python` in CI. I can make a quick PR to update it.

**Comment by nkoep** [comment]
> @nkoep Do you think we should try and add a `pip install -U pip setuptools wheel` in the CI ? This repository does use the older version of `setup-python` in CI. I can make a quick PR to update it. Or maybe update this PR

I think this should be enough:

```diff
diff --git a/pyproject.toml b/pyproject.toml
index bcb5f7e..5f885e4 100644
--- a/pyproject.toml
+++ b/pyproject.toml
@@ -1,5 +1,5 @@
 [build-system]
-requires = ["pip==21.3.1", "setuptools==59.6.0", "setuptools-scm[toml]>=6.2"]
+requires = ["pip==22.3.1", "setuptools==65.6.3", "setuptools-scm[toml]>=6.2"]
 build-backend = "setuptools.build_meta"
 ```
Feel free to add it to this PR.

**Comment by SauravMaheshkar** [comment]
Can I add 

```lang-toml
version = "2.1.1"
```

File: pyproject.toml [hunk 6]
```diff
@@ -1,10 +1,101 @@
 [build-system]
-requires = ["pip==21.3.1", "setuptools==59.6.0", "setuptools-scm[toml]>=6.2"]
+requires = ["pip==22.3.1", "setuptools==65.6.3", "setuptools-scm[toml]>=6.2"]
 build-backend = "setuptools.build_meta"
 
+[project]
+name = "pymanopt"
```
Comment (nkoep) [inline pyproject.toml:6]: This seems necessary to get `setuptools-scm` to work.
```suggestion
name = "pymanopt"
dynamic = ["version"]
```

**Pull request merged**